src/domain/fiat/index.types.d.ts

```diff
@@ -0,0 +1 @@
+type FiatAmount = number & { readonly brand: unique symbol }
```

Comment:
I have doubts about `domain/fiat` as a useful grouping. `fiat` isn't an entity or aggregate in the system which is typically the boundary for which behaviour is defined.
We do have the `bitcoin/{onchain,offchain}` modules which define a bunch of primitives for handling bitcoin stuff - perhaps it is, or will become something similar. But until it becomes more than a single primitive type I would probably group it somewhere else.

Reply:
we'll have some logic in here shortly with the dealer. I have been thinking of naming it dealer instead of fiat otherwise, any thoughts on this?